tools/expressionmerge.py

```diff
@@ -35,7 +36,7 @@
     base, ext = os.path.splitext(f)
     delimiter = ';' if ext == '.csv' else '\t'
 
-    with gzip.open(f, 'rb') as csvfile:
+    with io.TextIOWrapper(io.BufferedReader(gzip.open(f))) as csvfile:
```

Comment:
> @kostko, is there a better way? The problem is that users upload .csv files with windows (/r/n), linux (/n) or mac (/r) line endings and that simply iterating through the file (gzip.open) fails for .csv files generated with Excel on Macs.

What do you mean that simply iterating through the file fails? You mean that the CSV parser does not handle different line endings properly?

> @kostko, would you suggest a quick linux command that checks line endings and convert them from win or mac to a linux style? One option would be to do this for all files after users upload them to the platform.

There is the `dos2unix`, which does this conversion.

> Does io.TextIOWrapper(io.BufferedReader(...)) impact performance? Can we use this for really large files as well? 

There certainly is some overhead, but since it is a stream interface, it might be very low. The best way would be to benchmark its impact with some large files using a simple Python script.